src/basic/strv.h

```diff
@@ -38,6 +38,9 @@ char** strv_copy_n(char * const *l, size_t n);
 static inline char** strv_copy(char * const *l) {
         return strv_copy_n(l, SIZE_MAX);
 }
+static inline char** strv_copy_or_null(char * const *l) {
+        return l ? strv_copy_n(l, SIZE_MAX) : NULL;
```

Comment:
This would mean we cannot distinguish the ENOMEM case from the empty strv case. 

That's why I proposed a signature like `int strv_copy_unless_empty(char **list, char ***ret)`.

That way we can just do `if (r < 0) return r` to catch failures from this call